backpack/core/derivatives/batchnorm_nd.py

```diff
@@ -0,0 +1,312 @@
+"""Contains derivatives for BatchNorm."""
+from typing import List, Tuple, Union
+
+from torch import Size, Tensor, einsum
+from torch.nn import BatchNorm1d, BatchNorm2d, BatchNorm3d
+
+from backpack.core.derivatives.basederivatives import BaseParameterDerivatives
+
+
+class BatchNormNdDerivatives(BaseParameterDerivatives):
+    """Derivatives for BatchNorm.
+
+    If training=False: saved statistics are used.
+    If training=True: statistics of current batch are used.
+
+    Index convention:
+    n: batch axis
+    c: category axis
+    {empty}/l/hw/dhw: dimension axis for 0/1/2/3-dimensions
+
+    As a starting point, see these references:
+    https://kevinzakka.github.io/2016/09/14/batch_normalization/
+    https://chrisyeh96.github.io/2017/08/28/deriving-batchnorm-backprop.html
+    """
+
+    def __init__(self, n_dim: int):
+        """Initialization.
+
+        Args:
+            n_dim: number of dimensions
+        """
+        self.n_dim = n_dim
```

Comment:
Rename `n_dim` -> `N` to keep consistent with other `*NdDerivatives`?

Reply:
For consistency, I agree. However I chose `n_dim`, because at the moment we are using `N` for
- `batch_size`
- number of free dimensions
If we have a good name for the second one, it would be better than `N`.
Do you have a suggestion other than `n_dim`?
Should I just change it to `N` for consistency?